zapisy/apps/enrollment/records/models/records.py

```diff
@@ -223,14 +223,31 @@ def list_waiting_students(
             status=RecordStatus.QUEUED, group__course__in=courses).values(
                 'group__course', 'group__type', 'student__user',
                 'student__user__first_name', 'student__user__last_name')
+                
+        to_combine = Record.objects.filter(
+            status=RecordStatus.QUEUED, group__course__in=courses).values(
+                'group__course', 'group__id', 'group__type', 'student__user',
+                'student__user__first_name', 'student__user__last_name')
+        
         enrolled = Record.objects.filter(
             status=RecordStatus.ENROLLED, group__course__in=courses).values(
                 'group__course', 'group__type', 'student__user', 'student__user__first_name',
                 'student__user__last_name')
+       
         waiting = queued.difference(enrolled)
+        
+        wt = []
+        
+        for query in waiting:
+        	for query_all in to_combine:
+        		if ( (query['group__course']==query_all['group__course']) and (query['group__course']==query_all['group__course']) and (query['group__type']==query_all['group__type']) and (query['student__user']==query_all['student__user']) and (query['student__user__first_name']==query_all['student__user__first_name']) and (query['student__user__last_name']==query_all['student__user__last_name']) ):
```

Comment:
To taki cross-join zrobiony ręcznie w czasie kwadratowym (zamiast _n lg n_)?

```suggestion
        		if ( (query['group__course']==query_all['group__course']) and (query['group__course']==query_all['group__course']) and (query['group__course']==query_all['group__course']) and (query['group__course']==query_all['group__course']) and (query['group__type']==query_all['group__type']) and (query['student__user']==query_all['student__user']) and (query['student__user__first_name']==query_all['student__user__first_name']) and (query['student__user__last_name']==query_all['student__user__last_name']) ):
```